### ISSUE TITLE
Can't build with GCC 6

### ISSUE THREAD

**Issue by Titan-C** [TRIQS/triqs#328]
Maybe is cmake that does not understand the new version of GCC anthough is supported. I get this message 
```
-- Installation directory is /home/oscar/miniconda3/envs/dev2
-- Compiler is gcc with version g++ (GCC) 6.1.1 20160501
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software see the source for copying conditions.  There is NO
warranty not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
CMake Error at cmake/FindCompilerCheck.cmake:79 (MESSAGE):
  

  ************************** FATAL ERROR ************************************

  You are using the gcc compiler but your compiler is too old :

   TRIQS requires version >= 4.8.1 while you have g++ (GCC) 6.1.1 20160501

  Copyright (C) 2016 Free Software Foundation, Inc.

  This is free software see the source for copying conditions.  There is NO

  warranty not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

    

  ************************** FATAL ERROR ************************************

Call Stack (most recent call first):
  CMakeLists.txt:62 (find_package)


-- Configuring incomplete, errors occurred!

```

**Comment by parcollet** [comment]
yes, it does compile on gcc 6 normally.
this is the simple cmake check ...
what does g++ --version says exactly ?

**Comment by parcollet** [comment]
which distribution by the way  ?

**Comment by Titan-C** [comment]
Olivier Parcollet <notifications@github.com> writes:

> what does g++ --version says exactly ?

oscar@titan-arch-m ~ $g++ --version
g++ (GCC) 6.1.1 20160501
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

> which distribution by the way ?

Is Arch linux

**Comment by parcollet** [comment]
ok, the check of version is not nicely done. I will patch this.

**Comment by parcollet** [comment]
please recheck on branch gcc6, I have pushed a little patch.
Need to recheck on OS X, etc.. before merging.

**Comment by Titan-C** [comment]
Yes it builds.
One question, isn't it simpler to use cmake to check the version instead of using and extra compiled c++ file?
Something like:
```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.1")
    message(FATAL_ERROR "Insufficient gcc version")
  endif()
# etc.
endif()
```

**Comment by parcollet** [comment]
It is simpler. I did not know this variable, it is portable on every platform (OS X, linux ...) ?

**Comment by Titan-C** [comment]
I don't know about portability. But as a CMAKE variable it shall work wher CMAKE works.

**Comment by aichhorn** [comment]
Thanks Igor for referencing this. Obviously, looking at the fix, it never made it into the 1.3 master...
